Feat: #1 Create 회원 가입 기능

hongggs opened the pull request:
- 회원 가입 요청을 처리하는 DTO(MemberSignupRequest) 추가
- MemberService에 회원 가입 비즈니스 로직 구현
- 비밀번호 검증을 위한 PasswordValidator 유틸리티 클래스 추가
- 회원 가입 컨트롤러(MemberController) 생성 및 /signup 엔드포인트 추가
- MemberRepository에 회원 중복 체크를 위한 메서드 추가

수정할 부분 관련해 말씀해 주시면 바로 반영하겠습니다.

hongggs committed:
Feat: #1 Add 회원가입 중복 체크, 비밀번호 정책, 유효성 검사 관련 ErrorCode 추가

- 회원가입 시 이메일 중복 체크를 위한 EMAIL_ALREADY_EXISTS ErrorCode 추가
- 회원가입 시 계정 중복 체크를 위한 ACCOUNT_ALREADY_EXISTS ErrorCode 추가
- 비밀번호 정책 위반 시 사용되는 PASSWORD_VIOLATES_POLICY ErrorCode 추가
- 유효하지 않은 입력 값에 대한 INVALID_INPUT_VALUE ErrorCode 추가

hongggs committed:
Feat: #1 Add 유효성 검사 예외 처리 로직 GlobalExceptionHandler에 추가

- MethodArgumentNotValidException 처리를 위한 handleValidationExceptions 메서드 추가
- 유효성 검사 실패 시 INVALID_INPUT_VALUE ErrorCode와 결합된 오류 메시지 반환

hongggs committed:
Chore: #1 Add dto 유효성 검증을 위한 의존성

Perhona committed:
Feat: #9 Add CI/CD 환경 설정 및 헬스체크 컨트롤러 추가

- application.yml 파일에 blue, green, prod 환경 설정 추가
- logback-prod.xml 로깅 파일 설정 변경
- Dockerfile 추가로 컨테이너 환경 설정
- HealthCheckController 헬스체크 API 추가

Minerva08 committed:
Chore: Add QueryDsl 파일 경로 .gitignore 추가
- /src/main/generated/

hongggs committed:
Feat: #1 Create 회원 가입 기능

- 회원 가입 요청을 처리하는 DTO(MemberSignupRequest) 추가
- MemberService에 회원 가입 비즈니스 로직 구현
- 비밀번호 검증을 위한 PasswordValidator 유틸리티 클래스 추가
- 회원 가입 컨트롤러(MemberController) 생성 및 /signup 엔드포인트 추가
- MemberRepository에 회원 중복 체크를 위한 메서드 추가

Minerva08 commented on src/main/generated/com/feedlink/api/feedlink_api/domain/hashtag/entity/QHashtag.java:
```diff
@@ -0,0 +1,42 @@
+package com.feedlink.api.feedlink_api.domain.hashtag.entity;
```
git ignore에 추가 되어 있습니다 
변경사항이 있어 dev pull 부탁드려요

hongggs replied:
넵 알겠습니다!

Minerva08 commented:
dev pull 하고 머지 하시면 될것 같습니다~

Perhona commented on src/main/java/com/feedlink/api/feedlink_api/domain/member/service/MemberService.java:
```diff
@@ -0,0 +1,77 @@
+        // 계정 중복 검사
+        if (memberRepository.findByMemberAccount(request.getMemberAccount()).isPresent()) {
+            throw new CustomException(ErrorCode.ACCOUNT_ALREADY_EXISTS);
+        }
```
현재 메일 중복 검사, 계정 중복 검사를 위해 조회 쿼리가 두 번 발생하고 있는데, 아래처럼 한 번의 조회로 해볼 수도 있을 것 같아요!
```javax
Optional<Member> memberOptional = memberRepository.findByMemberEmailOrMemberAccount(request.getMemberEmail(), request.getMemberAccount());

    if (memberOptional.isPresent()) {
        Member existingMember = memberOptional.get();

        if (existingMember.getMemberEmail().equals(request.getMemberEmail())) {
            throw new CustomException(ErrorCode.EMAIL_ALREADY_EXISTS);
        }

        if (existingMember.getMemberAccount().equals(request.getMemberAccount())) {
            throw new CustomException(ErrorCode.ACCOUNT_ALREADY_EXISTS);
        }
    }
```

hongggs replied:
오 이렇게 하면 쿼리 조회가 한 번 발생하겠네요!! 수정하겠습니다~

Perhona commented on src/main/java/com/feedlink/api/feedlink_api/domain/member/dto/MemberSignupRequest.java:
```diff
@@ -0,0 +1,25 @@
+import lombok.Setter;
+
+@Getter
+@Setter
```
setter는 어떤 경우에 사용되는지 궁금합니다!

hongggs replied:
확인해보니 필요하지 않은 어노테이션이라 삭제하겠습니다~

hongggs committed:
Refactor: Delete 필요없는 Setter 어노테이션

hongggs committed:
Merge branch 'dev' into feature/hs_#1_join

hongggs committed:
Refactor: #1 Modify 회원 가입 시 중복 검사 쿼리

- 이메일 중복 검사와 계정 중복 검사를 하나의 조회 쿼리로 통합
- MemberRepository에 findByMemberEmailOrMemberAccount 메서드 추가
- 조회된 회원 정보를 바탕으로 중복 여부를 검증하도록 로직 수정
